src/tracing/js/builtins.rs

```diff
@@ -34,7 +34,11 @@ pub(crate) fn to_serde_value(val: JsValue, ctx: &mut Context) -> JsResult<serde_
             )
         })
     } else {
-        val.to_json(ctx)
+        val.to_json(ctx)?.ok_or_else(|| {
```

Comment:
returns None for undefined, looks fine https://docs.rs/boa_engine/latest/boa_engine/value/struct.JsValue.html#method.to_json